### PR TITLE
FOG-408: Make ingest peer uri filtering based on responder id

### PR DESCRIPTION
**Pull request by garbageslam**
This fixes a bug that exists today in fog ingest deployments.
I thought this was only a minor issue because, it would wrongly
add the peer listen uri to the list of peer uris, but then filter
it out and not connect to it. But because the active server replicates
its peer list onto the backups continuously, they will all end up
with this uri in their list, and there is no configuration that will
fix this. Additionally, a server won't be able to filter out it's
own (real) URI that the peers have in their lists, because it doesn't
filter by responder id before this commit.

### CODE REVIEW

**Comment by sugargoat** [comment]
Was there a successful deployment with this revision?

**Comment by garbageslam** [comment]
not yet

**Review by eranrund — COMMENTED** [review]
LGTM once everything is green and happy. Let me know if you want to offload getting it there to me.

**Review by sugargoat — APPROVED** [review]
LGTM once successful deploy

**Review by jcape — APPROVED** [review]
It's fine as-is, but it really seems like the "is the responder ID valid" test should happen when creating an `IngestPeerUri`, no? Is there ever a case where an `IngestPeerUri` without a responder ID is valid?

**Comment by garbageslam** [comment]
no, i did not create the responder id API -- i made a ticket already to track making responder id infallible

File: fog/ingest/server/src/config.rs [hunk 1]
```diff
@@ -38,7 +38,7 @@ pub struct IngestConfig {
     pub peer_listen_uri: IngestPeerUri,
 
     /// List of all peers in this cluster
-    #[structopt(long)]
+    #[structopt(long, use_delimiter = true)]
```
Comment (eranrund) [inline fog/ingest/server/src/config.rs:41]: Why was this needed?

Comment (garbageslam) [reply fog/ingest/server/src/config.rs:41]: if you want to have it work like `--peers igp://foo,igp://bar,igp://baz,igp://qaz` you need to put `use_delimter = true`. Otherwise, when structopt sees a Vec, you must make it like `--peers igp://foo --peers igp://baz --peers igp://qaz` and that just seems harder to read

Comment (garbageslam) [reply fog/ingest/server/src/config.rs:41]: here's the issue in structopt repo: https://github.com/TeXitoi/structopt/issues/94
